### PR TITLE
HighLevel transfer methods, associated changes(Chunk Reader/Writer) and test cases

### PR DESCRIPTION
**Pull request by mohsha-msft**
- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] Apache v2 license headers are included in each file.
 
[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/master/CHANGELOG.md

### CODE REVIEW

File: sdk/storage/azblob/highlevel.go [hunk 1]
```diff
@@ -0,0 +1,641 @@
+package azblob
```
Comment (adreed-msft) [inline sdk/storage/azblob/highlevel.go:1]: I feel like highlevel's port and the tests should be separate PRs.

File: sdk/storage/azblob/highlevel.go [hunk 2]
```diff
@@ -0,0 +1,646 @@
...
+	"bytes"
+	"errors"
+	"os"
+	//"github.com/Azure/azure-pipeline-go/pipeline"
+)
+
+//// CommonResponse returns the headers common to all blob REST API responses.
+type CommonResponse interface {
+	//// ETag returns the value for header ETag.
+	//ETag() *string
```
Comment (zezha-msft) [inline sdk/storage/azblob/highlevel.go:21]: These don't apply anymore, to remove.

File: sdk/storage/azblob/highlevel.go [hunk 3]
```diff
@@ -0,0 +1,646 @@
...
+	//
+	//// Version returns the value for header x-ms-version.
+	//Version() string
+
+	// Response returns the raw HTTP response object.
+	Response() *http.Response
+}
+
+// HighLevelUploadToBlockBlobOption identifies options used by the UploadBufferToBlockBlob and UploadFileToBlockBlob functions.
+type HighLevelUploadToBlockBlobOption struct {
```
Comment (zezha-msft) [inline sdk/storage/azblob/highlevel.go:40]: To follow up later: the keyword "HighLevel" seems a bit off, ideally these struct names should align with func that takes them in.

**Review by zezha-msft — APPROVED** [review]

**Comment by zezha-msft** [comment]
We are forced to do a squash merge here.
